src/agent/src/main.rs

```diff
@@ -384,6 +400,10 @@ async fn start_sandbox(
     let (tx, rx) = tokio::sync::oneshot::channel();
     sandbox.lock().await.sender = Some(tx);
 
+    if Path::new(CDH_PATH).exists() && Path::new(AA_PATH).exists() {
+        init_attestation_components(logger)?;
```

Comment:
Unless I'm missing something, the way attestation components are launched will have a huge impact on start time of sandbox. Each call to `launch_process` will spend up to 6 seconds on ` wait_for_path_to_exist`... I think attestation components should be started asynchronously just like the policy agent.

Has this been discussed before?

Reply:
Yeah we should probably look for some optimization here, but it's a bit tricky because we will need some of these services to be running before we can do certain things (like decrypt a container). Maybe we could at least try to start all three guest components in parallel. That might be for a followup PR tho.

Reply:
I'm not sure if starting them in parallel will work (and even what we have now might be flakey?) Magnus found in peer pods that ASR needs the socket from CDH, so had to add it as a systemd dependency (in https://github.com/confidential-containers/cloud-api-adaptor/pull/1751), otherwise ASR had a chance of starting and immediately failing. It does feel like we have the answer there rather than coming up with our own solution, so I'm not proposing that it's worth creating our own system services start-up logic in this PR!

Reply:
Hi @stevenhorsman @fitzthum , in my first comment I wasn't actually thinking in parallelizing ASR/CDR/AA to speed up the boot of attestation components, rather I was concerned about the kata-agent main thread having to wait on attestation components. Now I understood that kata-agent cannot be in ready state prior to all attestation components.

This still can hurt badly the pod's startup time. We might not address that issue on this PR but I think we still need to draw solutions to ensure that any follow on fix won't put us on a bad spot.

Reply:
I'm going to resolve this thread as I believe it won't be implemented on this PR.